### PR TITLE
Add blog post: Memory Management improvements for Flink’s JobManager in Apache Flink 1.11

### PR DESCRIPTION
**Pull request by azagrebin**
Source in [gdoc](https://docs.google.com/document/d/1K7Kte14C6txZdiWNIfLwD7NVbnEKwHzWjLtcNllwQBk).

### CODE REVIEW

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title: "Memory Management improvements for Flink’s JobManager in Apache Flink 1.11"
+date: 2020-09-01T15:30:00.000Z
+authors:
+- Andrey:
+  name: "Andrey Zagrebin"
+categories: news
+excerpt: In a previous blog post focused on the memory model of the TaskManagers and how it was improved with the Apache Flink 1.10 release. This blog post addresses the same topic but for the JobManager instead.
```
Comment (MarkSfik) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
excerpt: In a previous blog post, we focused on the memory model of the TaskManagers and its improvements with the Apache Flink 1.10 release. This blog post addresses the same topic but for Flink's JobManager instead.
```

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+Apache Flink 1.11 comes with significant changes to the memory model of Flink’s JobManager and configuration options for your Flink clusters. These recently-introduced changes make Flink adaptable to all kinds of deployment environments (e.g. Kubernetes, Yarn, Mesos), providing better control over its memory consumption.
+
+The [previous blog post]({{ site.baseurl }}/news/2020/04/21/memory-management-improvements-flink-1.10.html), focused on the memory model of the TaskManagers and how it was improved with the Apache Flink 1.10 release. This blog post addresses the same topic but for the JobManager instead. Flink 1.11 unifies the memory model of Flink’s processes. The newly introduced memory model of the JobManager follows a  similar approach to that of the TaskManagers; it is simpler and has fewer components and tuning knobs. This post might then seem very similar to our previous story on Flink’s memory, but aims at providing a complete overview of Flink’s JobManager memory model as of Flink 1.11. Read on for a full list of updates and changes below!
```
Comment (MarkSfik) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
The [previous blog post](https://flink.apache.org/news/2020/04/21/memory-management-improvements-flink-1.10.html), focused on the memory model of the TaskManagers and how it was improved in Flink 1.10. This post addresses the same topic but for the JobManager instead. Flink 1.11 unifies the memory model of Flink’s processes. The newly-introduced memory model of the JobManager follows a similar approach to that of the TaskManagers; it is simpler and has fewer components and tuning knobs. This post might consequently seem very similar to our previous story on Flink’s memory but aims at providing a complete overview of Flink’s JobManager memory model as of Flink 1.11. Read on for a full list of updates and changes below!
```

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+Having a clear understanding of Apache Flink’s process memory model allows you to manage resources for the various workloads more efficiently. The following diagram illustrates the main memory components of a Flink process:
+
+<center>
+<img src="{{ site.baseurl }}/img/blog/2020-09-01-flink-1.11-memory-management-improvements/total-process-memory-flink-1.11.png" width="400px" alt="Backpressure sampling:high"/>
+<br/>
+<i><small>Flink: Total Process Memory</small></i>
+</center>
+<br/>
+
+The JobManager process is a JVM process. On a high level, its memory consists of the JVM Heap and Off-Heap memory. These types of memory are consumed by Flink directly or by JVM for its specific purposes (i.e. metaspace etc). There are two major memory consumers within the JobManager process: the framework itself consuming memory for internal data structures, network communication, etc. and the user code which runs within the JobManager process, e.g. in certain batch sources or in checkpoint completion callbacks.
```
Comment (MarkSfik) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
The JobManager process is a JVM process. On a high level, its memory consists of the JVM Heap and Off-Heap memory. These types of memory are consumed by Flink directly or by JVM for its specific purposes (i.e. metaspace). There are two major memory consumers within the JobManager process: the framework itself consuming memory for internal data structures, network communication, etc. and the user code which runs within the JobManager process, e.g. in certain batch sources or during checkpoint completion callbacks.
```

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+<div class="alert alert-info" markdown="1">
+<span class="label label-info" style="display: inline-block"><span class="glyphicon glyphicon-info-sign" aria-hidden="true"></span> Note</span>
+Please, note that the user code has direct access to all memory types: JVM Heap, Direct and Native memory. Therefore, Flink cannot really control its allocation and usage.
```
Comment (MarkSfik) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
Please note that the user code has direct access to all memory types: JVM Heap, Direct and Native memory. Therefore, Flink cannot really control its allocation and usage.
```

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+* **[JVM requires more internal memory](https://ci.apache.org/projects/flink/flink-docs-release-1.11/ops/memory/mem_setup_jobmanager.html#detailed-configuration).** There is no direct control over certain types of JVM process allocations but Flink provides _JVM Overhead_ options. The _JVM Overhead_ options allow declaring an additional amount of memory, anticipated for those allocations and not covered by other options.
+
+## Conclusion
+
+The latest Flink release ([Flink 1.11](https://flink.apache.org/downloads.html#apache-flink-1111)) introduces some significant changes to the memory configuration of Flink’s JobManager, making Flink’s memory management significantly easier than before. Stay tuned for more additions and features in upcoming releases. If you have any suggestions or questions for the community, we encourage you to sign up to the Apache Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) and become part of the discussion.
```
Comment (MarkSfik) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
The latest Flink release ([Flink 1.11](https://flink.apache.org/downloads.html#apache-flink-1111)) introduces some notable changes to the memory configuration of Flink’s JobManager, making its memory management significantly easier than before. Stay tuned for more additions and features in upcoming releases. If you have any suggestions or questions for the Flink community, we encourage you to sign up to the Apache Flink [mailing lists](https://flink.apache.org/community.html#mailing-lists) and become part of the discussion.
```

**Review by MarkSfik — COMMENTED** [review]
Thanks for the article @azagrebin 
This is a great contribution highlighting one of the features of the Flink 1.11  release. 

I left some minor suggestions in the document. 
Let me know if you have any questions. 
Markos

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md (hunk 2 above)
Comment (XComp) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
The [previous blog post]({{ site.baseurl }}/news/2020/04/21/memory-management-improvements-flink-1.10.html) focused on the memory model of the TaskManagers and how it was improved with the Apache Flink 1.10 release. This blog post addresses the same topic but for the JobManager instead. Flink 1.11 unifies the memory model of Flink’s processes. The newly introduced memory model of the JobManager follows a similar approach to that of the TaskManagers; it is simpler and has fewer components and tuning knobs. This post might seem very similar to our previous story on Flink’s memory, but aims at providing a complete overview of Flink’s JobManager memory model as of Flink 1.11. Read on for a full list of updates and changes below!
```

Comment (XComp) [reply _posts/2020-09-01-flink-1.11-memory-management-improvements.md]: Ok, it's hard to see in the diff above: I removed the comma in the first sentence of this paragraph.

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md (hunk 3 above)
Comment (XComp) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
The JobManager process is a JVM process. On a high level, its memory consists of the JVM Heap and Off-Heap memory. These types of memory are consumed by Flink directly or by the JVM for its specific purposes (i.e. metaspace etc). There are two major memory consumers within the JobManager process: the framework itself consuming memory for internal data structures, network communication, etc. and the user code which runs within the JobManager process, e.g. in certain batch sources or in checkpoint completion callbacks.
```

Comment (XComp) [reply _posts/2020-09-01-flink-1.11-memory-management-improvements.md]: I don't see the difference, again. -.- Hence, here's what I changed: I added `the` in the second sentence: "These types of memory are consumed by Flink directly or by ***the*** JVM for [...]"

**Review by XComp — APPROVED** [review]
I read through it again. Nothing to add here.

**Comment by azagrebin** [comment]
Thanks for the review @MarkSfik @XComp 
I have addressed the comments

File: _posts/2020-09-01-flink-1.11-memory-management-improvements.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "Memory Management improvements for Flink’s JobManager in Apache Flink 1.11"
+date: 2020-09-01T15:30:00.000Z
+authors:
+- Andrey:
+  name: "Andrey Zagrebin"
+categories: news
```
Comment (morsapaes) [inline _posts/2020-09-01-flink-1.11-memory-management-improvements.md]:
```suggestion
```

Comment (morsapaes) [reply _posts/2020-09-01-flink-1.11-memory-management-improvements.md]: Please don't use this category for regular blogposts.

**Comment by morsapaes** [comment]
@azagrebin, there are a few extra files committed that shouldn't be there. Would you mind removing anything that is not 1) your blogpost **.md** file and 2) any **image** files?

**Comment by azagrebin** [comment]
Thanks for the review @morsapaes 
I removed  `categories: news` and the rebuild commit with non-md/image files

**Pull request merged**